homeassistant/components/profiler/__init__.py

```diff
@@ -1,22 +1,47 @@
 """The profiler integration."""
 import asyncio
 import cProfile
+from datetime import timedelta
+import logging
 import time
 
 from guppy import hpy
+import objgraph
 from pyprof2calltree import convert
 import voluptuous as vol
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_time_interval
 from homeassistant.helpers.service import async_register_admin_service
 from homeassistant.helpers.typing import ConfigType
 
 from .const import DOMAIN
 
 SERVICE_START = "start"
 SERVICE_MEMORY = "memory"
+SERVICE_START_LOG_OBJECTS = "start_log_objects"
+SERVICE_STOP_LOG_OBJECTS = "stop_log_objects"
```

Comment:
Instead of start + stop, should we do a start that automatically stops after X time? That's how we've implemented the MQTT dump too. Or is there a use case where people want to keep it running forever ?

Reply:
Usually this runs for a day at time to find slow leaks.  I currently have one running for ~4 days and I wouldn't want it to stop on me.